packages/cache-manager/index.ts

```diff
@@ -0,0 +1,42 @@
+import { CacheManagerOptions } from "./cache-manager-options";
+import { ICache } from "./cache-type";
+
+export type IDataProviderCallback = () => string | Promise<string>;
+
+export interface CacheManager {
+    getItem: (key: string) => Promise<string>;
+    setItem: (key: string, value: string) => Promise<void>;
+    wrap: (key: string, fallback: IDataProviderCallback) => Promise<string>;
+}
+
+export function createCacheManager(cacher: ICache, { ttl }: CacheManagerOptions = { ttl: 600 }): CacheManager {
+
+    return {
+        getItem: cacher.getItem,
+        setItem(key: string, value: string) {
+            return cacher.setItem(key, value, { ttl });
+        },
+        async wrap(key: string, fallback: IDataProviderCallback) {
+            let currentValue;
+            try {
+                currentValue = await cacher.getItem(key);
+            } catch {
+                //
```

Comment:
Don't we want to add error error message ? 
Something like : `couldn't get item from ${key}`

Reply:
the "wrap" operation wants to check if there's a valid value at the `cacher`.
if there isn't - it runs the fallback anyways, never mind if it's because of an exception of a `nullish` value.